src/r_things.c

```diff
@@ -834,6 +834,12 @@ void R_DrawPSprite (pspdef_t *psp)
   // [crispy] free look
   vis->texturemid += (centery - viewheight/2) * pspriteiscale;
 
+  // Prevent visual artifacts with blocky fuzz.
+  if (!vis->colormap[0] && blocky_fuzz)
+  {
+    vis->texturemid = (vis->texturemid >> FRACBITS) << FRACBITS;
```

Comment:
So, this removes the fractional part of `vis->texturemid`. What if we *always* do this? Well this be noticeable for non-fuzzed weapon sprites?

Reply:
I'm not sure. I'll take a closer look when I'm free.

Reply:
Thanks! 

Also, I think this would be easier to read. 

```c
vis->texturemid &= (FRACUNIT - 1) << FRACBITS;
```

Reply:
Okay, here's my best guess. To reduce the appearance of pixels jumping around, Doom uses subpixel-accurate texture mapping. That's why we see the fractional parts of values retained during intermediate calculations. In addition to level textures and thing sprites, this also applies to player (weapon) sprites. For player sprites, it's more noticeable with a reduced screen size. So the fractional part of `vis->texturemid` should not be discarded early.

For sprites, the fixed-point values are rounded in `R_DrawMaskedColumn()`, where the column bounds are set (`dc_yl` and `dc_yh`). `R_DrawFuzzColumn()` is called immediately after, where the fuzz effect is calculated and applied to the framebuffer.

The newest commit applies a fix in `R_DrawMaskedColumn()`. This means fuzzy thing sprites (e.g. spectres) also have a more accurate blocky appearance, in addition to the player sprites, as before.

It's probably more optimal to use a function pointer to a `R_DrawMaskedColumn_fuzz()` function instead of doing if-else every column in the loop.